packages/blocks/src/api/factory.js

```diff
@@ -407,17 +407,6 @@ export function switchToBlockType( blocks, name ) {
 	const firstBlock = blocksArray[ 0 ];
 	const sourceName = firstBlock.name;
 
-	// Unless it's a Grouping Block then for multi block selections
-	// check that all Blocks are of the same type otherwise
-	// we can't run a conversion
-	if (
-		! isContainerGroupBlock( name ) &&
-		isMultiBlock &&
-		! isBlockSelectionOfSameType( blocksArray )
-	) {
-		return null;
-	}
```

Comment:
I think initially, this was needed to limit the "paragraph -> list" conversion to paragraph only selections. It doesn't seem like the removal impacts this as we can't transform any multiselection to lists yet.

This is me saying, I agree with this change conceptually but how do you think we should define that a conversion is liimited to the same types or not?

Reply:
> how do you think we should define that a conversion is liimited to the same types or not?

What about adding a new property like `sameBlockTypes` to the transform?

```
transforms: {
	from: [
		{
			type: 'block',
			isMultiBlock: true,
			blocks: [ '*' ],
			sameBlockTypes: true
			__experimentalConvert( blocks ) { // ... },
		},
	},
}
```

`switchToBlockType` can then check the value of that property before applying the conversion:

```diff
diff --git a/packages/blocks/src/api/factory.js b/packages/blocks/src/api/factory.js
index 8775b6f78e..762a5c4d15 100644
--- a/packages/blocks/src/api/factory.js
+++ b/packages/blocks/src/api/factory.js
@@ -449,6 +449,9 @@ export function switchToBlockType( blocks, name ) {
 	let transformationResults;
 
 	if ( transformation.isMultiBlock ) {
+		if ( transformation.sameBlockTypes && ! isBlockSelectionOfSameType( blocksArray ) ) {
+			return null;
+		}
 		if ( has( transformation, '__experimentalConvert' ) ) {
 			transformationResults = transformation.__experimentalConvert(
 				blocksArray
```

Reply:
Naming is always difficult but I like the idea @mmtr !